rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -141,24 +146,33 @@ public synchronized void addNode(Node node) {
         synchronized (node.getNodeInformation().getURL().intern()) {
             // unknown host => start pinging process
             NodeSet toPing = new NodeSet();
+            HashMap<InetAddress, Long> hostsTopology = new HashMap<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "hostsTopology" object should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)